### PR TITLE
Feat/support number boolean date input params

### PR DESCRIPTION
**Pull request by jlalmes**
Seeking some feedback. This PR monkey patches the TRPCRouter to support `number`, `boolean` & `Date` out of the box (`string` was already supported) for `queries` inputs & `pathParameters`. I hope this would improve `trpc-openapi` incremental adoption, but I'm not sure wether I should just close this PR & force just encourage users to use `z.preprocess` instead...?

```ts
// Before PR
.query('some-query', {
  meta: { openapi: {...} },
  input: z.object({ 
    offset: z.preprocess((arg) => {
      if (typeof arg === 'string') return parseInt(arg);
      return arg;
    }, z.number().min(0)), 
    limit: z.preprocess((arg) => {
      if (typeof arg === 'string') return parseInt(arg);
      return arg;
    }, z.number().min(10).max(50)),
  }),
  ...
})
```

```ts
// After PR
.query('some-query', {
  meta: { openapi: {...} },
  input: z.object({ 
    offset: z.number().min(0), 
    limit: z.number().min(10).max(50),
  }),
  ...
})
```

### CODE REVIEW

File: src/adapters/node-http/monkeyPatch.ts [hunk 1]
```diff
@@ -29,7 +38,103 @@ export const monkeyPatchVoidInputs = (appRouter: OpenApiRouter) => {
...
+        }
+        if (instanceofZodTypeLikeBoolean(shapeSchema)) {
+          zObject = zObject.merge(z.object({ [shapeKey]: zBoolean(shapeSchema) }));
+          return;
+        }
+        if (instanceofZodTypeLikeDate(shapeSchema)) {
+          zObject = zObject.merge(z.object({ [shapeKey]: zDate(shapeSchema) }));
+          return;
+        }
+      });
```
Comment (KATT) [inline src/adapters/node-http/monkeyPatch.ts:94]: `throw` if there's an unexpected type which is unsupported?

**Review by KATT — APPROVED** [review]
you probably want some e2e/integration test with this, but it seems pretty okay too

File: src/adapters/node-http/monkeyPatch.ts (hunk 1 above)
Comment (jlalmes) [reply src/adapters/node-http/monkeyPatch.ts:94]: These inputs are already validated here: https://github.com/jlalmes/trpc-openapi/blob/9ed633cf596e42dc9c00501d383a6c68d14f1ab0/src/generator/schema.ts#L68

**Review by jqhoogland — CHANGES_REQUESTED** [review]
Awesome — this works great for what I needed it for!

My one complaint (if I'm allowed one) is that I find it very difficult to read those type guards, which probably makes errors likelier.

I think with the addition of a generic in types and a change in the type guards to mirror the generic structure would do a lot to make the new code more readable.

**Comment by jlalmes** [comment]
Closing as this PR is now very far out-of-date. This feature request is not forgotten about though, please see the open issue here: https://github.com/jlalmes/trpc-openapi/issues/44

**Pull request closed without merging**